stl/modules/std.compat.ixx

```diff
@@ -0,0 +1,532 @@
+
+// <cerrno>
+// Nothing to do, only macros.
+
+// <cfenv>
+export using std::fegetround;
+export using std::fesetround;
+export using std::feclearexcept;
+export using std::fegetenv;
+export using std::fegetexceptflag;
+export using std::feholdexcept;
+export using std::fenv_t;
+export using std::fesetenv;
+export using std::fesetexceptflag;
+export using std::fetestexcept;
+export using std::fexcept_t;
+export using std::feraiseexcept;
+export using std::feupdateenv;
+
+// <cfloat>
+// Nothing to do, only macros.
+
+// <cinttypes>
+export using std::imaxdiv_t;
+export using std::imaxabs;
+export using std::imaxdiv;
+export using std::strtoimax;
+export using std::strtoumax;
+export using std::wcstoimax;
+export using std::wcstoumax;
+
+// <climits>
+// Nothing to do, only macros.
+
+// <clocale>
+export using std::lconv;
+export using std::localeconv;
+export using std::setlocale;
+
+// <cmath>
+export using std::abs;
+export using std::acos;
+export using std::asin;
+export using std::atan;
+export using std::atan2;
+export using std::ceil;
+export using std::cos;
+export using std::cosh;
+export using std::exp;
+export using std::fabs;
+export using std::floor;
+export using std::fmod;
+export using std::frexp;
+export using std::ldexp;
+export using std::log;
+export using std::log10;
+export using std::modf;
+export using std::pow;
+export using std::sin;
+export using std::sinh;
+export using std::sqrt;
+export using std::tan;
+export using std::tanh;
+export using std::acosf;
+export using std::asinf;
+export using std::atanf;
+export using std::atan2f;
+export using std::ceilf;
+export using std::cosf;
+export using std::coshf;
+export using std::expf;
+export using std::fabsf;
+export using std::floorf;
+export using std::fmodf;
+export using std::frexpf;
+export using std::ldexpf;
+export using std::logf;
+export using std::log10f;
+export using std::modff;
+export using std::powf;
+export using std::sinf;
+export using std::sinhf;
+export using std::sqrtf;
+export using std::tanf;
+export using std::tanhf;
+export using std::acosl;
+export using std::asinl;
+export using std::atanl;
+export using std::atan2l;
+export using std::ceill;
+export using std::cosl;
+export using std::coshl;
+export using std::expl;
+export using std::fabsl;
+export using std::floorl;
+export using std::fmodl;
+export using std::frexpl;
+export using std::ldexpl;
+export using std::logl;
+export using std::log10l;
+export using std::modfl;
+export using std::powl;
+export using std::sinl;
+export using std::sinhl;
+export using std::sqrtl;
+export using std::tanl;
+export using std::tanhl;
+export using std::float_t;
+export using std::double_t;
+export using std::acosh;
+export using std::asinh;
+export using std::atanh;
+export using std::cbrt;
+export using std::erf;
+export using std::erfc;
+export using std::expm1;
+export using std::exp2;
+export using std::_Binary_hypot::hypot; // export only C's hypot(x, y), not C++'s hypot(x, y, z)

```

Comment:
[std.modules]/3 says: "The named module `std.compat` exports the same declarations as the named module `std`, and additionally exports declarations in the global namespace corresponding to the declarations in namespace `std` that are provided by the C++ headers for C library facilities ([tab:headers.cpp.c])."

This suggests to me that since `<cmath>` declares three-argument `hypot` in namespace `std`, this module should export three-argument `hypot` in the global namespace.

Reply:
After we talked about this on Discord, I've filed #3111 to clarify the Standardese here. The intended design was that `std.compat` shouldn't put things in the global namespace that the classic headers couldn't, and the current implementation follows that.